src/lib.rs

```diff
@@ -0,0 +1,436 @@
+    }
+
+    pub fn create_record(&self, bucket_id: String, collection_id: String,
+                         data: Option<JsonValue>) -> Result<JsonValue, KintoError> {
+        return self.create_record_with_perms(bucket_id, collection_id, data, None);
+    }
+
+    pub fn create_record_with_perms(&self, bucket_id: String, collection_id: String,
+                                    data: Option<JsonValue>,
+                                    permissions: Option<JsonValue>)
+                                    -> Result<JsonValue, KintoError> {
+
+        let endpoint = format!("{}/buckets/{bucket_id}/collections/{collection_id}/records",
+                               self.server_url, bucket_id=bucket_id,
+                               collection_id=collection_id);
+        let payload = self.set_payload(data, permissions);
+        return self.handle_request(Method::Post, endpoint, payload.into());
+    }
+
+    pub fn update_record(&self, bucket_id: String, record_id: String, collection_id: String,
+                         data: Option<JsonValue>) -> Result<JsonValue, KintoError> {
+        return self.update_record_with_perms(bucket_id, collection_id, record_id, data, None);
+    }
+
+    pub fn update_record_with_perms(&self, bucket_id: String, collection_id: String,
+                                    record_id: String, data: Option<JsonValue>,
+                                    permissions: Option<JsonValue>)
+                                    -> Result<JsonValue, KintoError> {
+
+        let endpoint = format!("{}/buckets/{bucket_id}/collections/{collection_id}/records/{id}",
+                               self.server_url, bucket_id=bucket_id,
+                               collection_id=collection_id, id=record_id);
+        let payload = self.set_payload(data, permissions);
+        return self.handle_request(Method::Post, endpoint, payload.into());
+    }
+
+    pub fn patch_record(&self, bucket_id: String, collection_id: String,
+                        record_id: String, data: Option<JsonValue>)
+                        -> Result<JsonValue, KintoError> {
+        return self.patch_record_with_perms(bucket_id, collection_id, record_id, data, None);
+    }
+
+    pub fn patch_record_with_perms(&self, bucket_id: String, collection_id: String,
+                                   record_id: String, data: Option<JsonValue>,
+                                   permissions: Option<JsonValue>)
+                                   -> Result<JsonValue, KintoError> {
+
+        let endpoint = format!("{}/buckets/{bucket_id}/collections/{collection_id}/records/{id}",
+                               self.server_url, bucket_id=bucket_id,
+                               collection_id=collection_id, id=record_id);
+        let payload = self.set_payload(data, permissions);
+        return self.handle_request(Method::Post, endpoint, payload.into());
+    }
+
+}
+
+
+pub fn prettyfy(data: JsonValue) -> String {
```

Comment:
Nit: I've seen `prettify` in other projects, e.g. https://github.com/google/code-prettify.